src/components/mc/cuda/mc_cuda.c

```diff
@@ -405,7 +406,7 @@ ucc_status_t ucc_ee_cuda_task_query(void *ee_req)
 {
     ucc_mc_cuda_stream_request_t *req = ee_req;
 
-    if (req->status != UCC_MC_CUDA_TASK_STARTED) {
```

Comment:
here ucc_assert(req->status != UCC_MC_CUDA_TASK_COMPLETED), right? It can be posted, completed_ack (in the non_blocking wait case) or started. started and completed_ack mean cuda task completes.

Reply:
correct, added assert